src/analyses/variable-sensitivity/value_set_abstract_object.cpp

```diff
@@ -46,12 +47,83 @@ class value_set_index_ranget : public index_ranget
   abstract_object_sett::const_iterator next;
 };
 
-index_range_ptrt make_value_set_index_range(
-  const value_set_abstract_objectt::abstract_object_sett &vals)
+index_range_ptrt make_value_set_index_range(const abstract_object_sett &vals)
 {
   return std::make_shared<value_set_index_ranget>(vals);
 }
 
+/// Determine abstract-type of an abstract object \p other.
+/// \param other: the abstract object to get the type of
+/// \return the abstract-type of \p other
+value_set_abstract_objectt::abstract_typet
+get_type(const abstract_object_pointert &other);
+
+/// Determine abstract-type of an expression-type \p type.
+/// \param type: the expression type to get the abstract-type of
+/// \return the abstract-type of \p type
+value_set_abstract_objectt::abstract_typet
+type_to_abstract_type(const typet &type);
```

Comment:
Am I correct in thinking that this is what picks which `abstract_objectt` subtype is used for the abstractions inside value set?

Reply:
Not exactly - it affects some runtime behaviour based on whether the value-set is holding pointers or values. This is related to the issue you highlighted above.